sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -182,4 +182,22 @@ object SQLExecution {
       body
     }(exec)
   }
+
+  /**
+   * Wrap passed function to ensure necessary thread-local variables like
+   * SparkContext local properties are forwarded to execution thread
+   */
+  def withThreadLocalCaptured[T](
+      sparkSession: SparkSession)(body: => T): Callable[T] = {
+    val activeSession = sparkSession
+    val sc = sparkSession.sparkContext
+    val localProps = Utils.cloneProperties(sc.getLocalProperties)
+    () => {
+      SparkSession.setActiveSession(activeSession)
```

Comment:
Please unset these once we you are done.

Reply:
done, i have rebased to master where this unset was already handled as part of separate JIRA